schema/policy.go

```diff
@@ -451,16 +419,11 @@ func (p *Policy) Check(action string, authorization Authorization, data map[stri
 		}
 	}
 
-	properties := p.Resource.Properties
-	if properties == nil {
```

Comment:
now this logic is missing

Reply:
I've restored this logic, but in my opinion this is unnecessary and only makes code uglier, since this check was only for debug purposes.

Reply:
I've run some performance tests to confirm that removing that logic doesn't hurt the performance. More details in the comments for this issue.